### PR TITLE
Dashboard component - add ui slice, expose selectors required for post messages

### PR DESCRIPTION
**Pull request by kandl**
- Add ui slice, so user can control relevant components from the outside
- Expose selectors required for post messages
- Minor fixes (do not expose event creators)

JIRA: RAIL-3385

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5236/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6949/ : FAILURE in 6m 50s

File: libs/sdk-backend-spi/src/common/settings.ts [hunk 1]
```diff
@@ -161,6 +161,11 @@ export interface ISettings {
      */
     enableCompanyLogoInEmbeddedUI?: boolean;
 
+    /**
+     * Indicates whether the insecure javascript events are enabled.
+     */
+    insecureJavascriptEventsEnabled?: boolean;
```
Comment (lupko) [inline libs/sdk-backend-spi/src/common/settings.ts:167]: i suggest not to publish this setting :)

File: libs/sdk-ui-dashboard/src/model/state/ui/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { uiInitialState } from "./uiState";
+import { uiReducers } from "./uiReducers";
+
+const uiSlice = createSlice({
+    name: "uiSlice",
+    initialState: uiInitialState,
+    reducers: uiReducers,
+});
+
+export const uiReducer = uiSlice.reducer;
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/model/state/ui/index.ts:12]: small stuff: please rename to `uiSliceReducer`.. it is sort of a convention with the other slices

File: libs/sdk-ui-dashboard/src/model/state/config/configSelectors.ts [hunk 3]
```diff
@@ -185,3 +185,39 @@ export const selectPlatformEdition = createSelector(selectConfig, (state) => {
...
+});
+
+/**
+ * Returns whether the insecure javascript events are enabled.
+ *
+ * @alpha
+ */
+export const selectInsecureJavascriptEventsEnabled = createSelector(selectConfig, (state) => {
+    return state.settings?.insecureJavascriptEventsEnabled ?? false;
+});
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/model/state/config/configSelectors.ts:205]: i suggest to remove this one; the entire settings can be selected and the flag tested in app

File: libs/sdk-ui-dashboard/src/model/events/index.ts [hunk 4]
```diff
@@ -244,20 +241,6 @@ export {
...
-    drillToCustomUrlRequested,
-    drillToDashboardRequested,
-    drillToInsightRequested,
-    drillToLegacyDashboardRequested,
-    drillDownResolved,
-    drillToAttributeUrlResolved,
-    drillToCustomUrlResolved,
-    drillToDashboardResolved,
-    drillToInsightResolved,
-    drillToLegacyDashboardResolved,
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/model/events/index.ts:260]: just for confirmation.. are you removing these in favor of command started events which Dan started last week or so?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/events/index.ts (hunk 4 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/events/index.ts:260]: These should not be public.

File: libs/sdk-ui-dashboard/src/model/state/config/configSelectors.ts (hunk 3 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/state/config/configSelectors.ts]: Removed

File: libs/sdk-ui-dashboard/src/model/state/ui/index.ts (hunk 2 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/state/ui/index.ts]: Renamed

File: libs/sdk-backend-spi/src/common/settings.ts (hunk 1 above)
Comment (kandl) [reply libs/sdk-backend-spi/src/common/settings.ts]: Removed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5238/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6951/ : SUCCESS in 14m 45s

**Review by lupko — COMMENTED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1757

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1495/ : FAILURE in 2m 40s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1772/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/562/ : SUCCESS in 14m 04s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6954/ : SUCCESS in 15m 18s

**Pull request merged**